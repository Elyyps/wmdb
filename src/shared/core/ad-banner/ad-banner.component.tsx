import * as React from "react";
import styles from "./ad-banner-component.module.scss";
import { Button } from "../button";
import { ILink } from "@app/api/core/link";

export interface IAdBannerComponentProps {
  advertisement: string;
  backgroundImage: string;
  button: ILink;
  content: string;
}

const AdBannerComponent = (props: IAdBannerComponentProps) => (
  <div className={styles["ad-banner"]} style={{ backgroundImage: `url(${props.backgroundImage})` }}>
    <div className={styles["ad-banner-item"]}>
      <div className={styles["ad-banner-title"]}>{props.content}</div>
      <Button title={props.button.title} href={props.button.url} />
    </div>
    <div className={styles["ad-banner-label"]}>{props.advertisement}</div>
  </div>
);

export { AdBannerComponent };
