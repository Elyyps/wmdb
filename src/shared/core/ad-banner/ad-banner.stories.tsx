import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { AdBannerComponent } from "./ad-banner.component";
import RECTANGLE from "@assets/rectangle.png";

storiesOf("AdBanner", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <AdBannerComponent
      content={"Surfen? Dat doe je in Zeeland."}
      backgroundImage={RECTANGLE}
      button={{ title: "Meer info", url: "/" }}
      advertisement={"Advertentie"}
    />
  ));
