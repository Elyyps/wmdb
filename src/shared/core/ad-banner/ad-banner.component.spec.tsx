import React from "react";
import { shallow } from "enzyme";
import { AdBannerComponent } from "./ad-banner.component";
import RECTANGLE from "@assets/rectangle.png";

describe("[AdBanner]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <AdBannerComponent
        content={"Surfen? Dat doe je in Zeeland."}
        backgroundImage={RECTANGLE}
        button={{ title: "Meer info", url: "/" }}
        advertisement={"Advertentie"}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
