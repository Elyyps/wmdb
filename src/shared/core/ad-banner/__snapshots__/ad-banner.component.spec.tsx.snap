// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AdBanner] should render component without crashing 1`] = `
<div
  className="ad-banner"
  style={
    Object {
      "backgroundImage": "url(rectangle.png)",
    }
  }
>
  <div
    className="ad-banner-item"
  >
    <div
      className="ad-banner-title"
    >
      Surfen? Dat doe je in Zeeland.
    </div>
    <Button
      href="/"
      title="Meer info"
    />
  </div>
  <div
    className="ad-banner-label"
  >
    Advertentie
  </div>
</div>
`;
